File: Euler10.js
const isPrime = n => {
	if (n % 2 == 0 && n != 2) {
		return false;
	}
	
	for ( let q=3 ; q<= n/2 ; q+=2 ) {
		if (n % q == 0) {
			return false;
		}
	}
	
	return true;
};

(function main() {
	let sum = 2;
	let currPrime = 0;
	
	for ( let q = 1 ; ; q+=2 ) {
		if (isPrime(q)) {
			currPrime = q;
			if (currPrime < 2000000) {
				sum += q;
				console.log(q);
			}
			else {
				break;
			}
		}
	}
	
	console.log(sum);
})();

File: euler10.py
#!/usr/bin/python3.4
from math import sqrt
from itertools import count, islice

sum = 2

def isPrime(num):
	return num > 1 and all(num % x for x in islice(count(2), int(sqrt(num)-1)))

for x in range(1,2000000,2):
	if isPrime(x):
		sum += x

print(sum)
	
